src/libsyntax_ext/diagnostics.rs

```diff
@@ -94,4 +94,38 @@ Considering that this would be a long explanation, we instead recommend you to
 take a look at the unstable book:
 https://doc.rust-lang.org/unstable-book/language-features/asm.html
 "##,
+
+E0665: r##"
+The `Default` trait was derived on an enum.
+
+Erroneous code example:
+
+```compile_fail,E0665
+#[derive(Default)]
+enum Food {
+    Sweet,
+    Salty,
+}
+```
+
+The `Default` cannot be derived on an enum for the simple reason that the
+compiler doesn't know which value to pick "by default" whereas it can for a
```

Comment:
I'm not sure the quotes are necessary here (or below).

Reply:
Sounded more funny. :D

Reply:
Agreed, I don't think the quotes add anything and I'd be in favor of removing them, but it's not that big of a deal. r=me either away